File: app/channels/live_cursor_channel.rb
class LiveCursorChannel < ApplicationCable::Channel
  def subscribed
    current_user.games.each do |game|
      stream_for game
    end
  end
  
  def unsubscribed
    stop_all_streams
  end
  
  def update_position(data)
    game_id = data['game_id'].to_i
    if game = current_user.games.find { |game| game.id == game_id }
      broadcast_to(game, data.merge('user' => {
        'id'   => current_user.id,
        'name' => current_user.name
      }))
    end
  end
end

File: app/javascript/channels/live_cursor_channel.js
import consumer from "./consumer"

const liveCursorChannel = consumer.subscriptions.create("LiveCursorChannel", {
  connected() {
    console.log('liveCursorChannel connected')
    // Called when the subscription is ready for use on the server
  },
  
  // Called when the subscription has been terminated by the server
  disconnected() {
    console.log('liveCursorChannel disconnected')
  },
  
  // Called when there's incoming data on the websocket for this channel
  received(data) {
    let element = document.querySelector(`[data-controller="map--cursor"][data-map--cursor-user-id="${data.user.id}"]`)
    if (element) {
      element.dataset['map-CursorX'] = data.position[0]
      element.dataset['map-CursorY'] = data.position[1]
      element.dataset['map-Floor']   = data.floor
      event = new Event('locationUpdated')
      element.dispatchEvent(event)
    }
  },
  
  sendPosition (gameId, floor, position) {
    this.perform('update_position', {
      game_id:  gameId,
      floor:    floor,
      position: [position.x, position.y]
    })
  }
});

export default liveCursorChannel
